pkgs/build-support/setup-hooks/auto-patchelf.py

```diff
@@ -264,10 +262,16 @@ def auto_patchelf(
     missing = [dep for dep in dependencies if not dep.found]
 
     # Print a summary of the missing dependencies at the end
+    print(f"auto-patchelf: {len(missing)} dependencies not be satisfied")
+    failure = False
     for dep in missing:
-        print(f"auto-patchelf could not satisfy dependency {dep.name} wanted by {dep.file}")
+        if dep.name.name in ignore_missing or "*" in ignore_missing:
+            print(f"warn: auto-patchelf ignoring missing {dep.name} wanted by {dep.file}")
+        else:
+            print(f"error: auto-patchelf could not satisfy dependency {dep.name} wanted by {dep.file}")
+            failure = True
 
-    if missing and not ignore_missing:
+    if failure:
         sys.exit('auto-patchelf failed to find all the required dependencies.\n'
                  'Add the missing dependencies to --libs or use --ignore-missing.')
```

Comment:
nit: maybe expand here on how to use ignore-missing properly ? Most people discover options by trial and error.

Reply:
good idea!

Reply:
Added some example usage in https://github.com/NixOS/nixpkgs/pull/167397/commits/a7fc2f6392be849f811c0a3f29f559311dc695df